apps/meteor/tests/e2e/README.md

```diff
@@ -35,3 +35,142 @@ Checking if a element is visible
 ```ts
  await expect(anyElement).toBeVisible();
  ```
+
+# Playwright Locator Best Practices
+
+## Preferred locator types:
+
+### 1. By role `.getByRole()`:
+This is the most recommended locator type. 
+- It ensures our element is accessible to screen readers and other assistive technologies.
+- It ensures that the element is uniquely identifiable.
+- It's recommended to use the `exact` option to ensure that the locator doesn't match any other element with the same role
+
+
+```ts
+await page.getByRole('button', { name: 'Save', exact: true });
+
+// Without the `exact` option, Playwright would match buttons with 'Save' and 'Save changes' labels.
+```
+
+
+### 2. By text `.getByText()` or `.getByLabel()`:
+- Use this when locating by role is not possible or not sufficient.
+- Be sure to restrict the scope of the locator to the element you are looking for.
+
+## Locator types to avoid (at all costs):
+If you are not able to find the element by role, label or text, it's a sign that the element is not accessible to the user.
+The component should be refactored to allow a more accessible locator.
+### 1. By `data-qa-id` or testId `.getByTestId()`:
+```ts
+// DON'T ❌ 
+page.locator('[data-qa-id="menu-more-actions"]'); 
+
+// DO ✅
+page.getByRole('menu', {name: 'More actions', exact: true }); 
+```
+
+### 2. By elements and class names `.locator('div.class-name')`:
+The HTML structure and/or class names can easily change over time, so it's better to use the locator by role and name.
+
+```ts
+// DON'T ❌
+page.locator('#modal-root .rcx-button-group--align-end .rcx-button--primary');
+
+// #modal-root was used to locate the injected portal element for the modal.
+// This can be better located by role 'dialog' and making sure the modal has the proper name attribute.
+
+// DO ✅
+page.getByRole('dialog', name: 'Modal name example').getByRole('button', { name: 'Confirm', exact: true });
+```
+
+### 3. By label with `has`:
+Our input elements hide the native input and render a custom component.
+
+To target the input and trigger changes, you should locate the label that wraps the input and use the `has` locator.
+```ts
+page.locator('label', { has: this.page.getByRole('checkbox', { name: 'Private' }) });
+```
+## Use our page-objects:
+> apps/meteor/tests/e2e/page-objects
+- page-objects are a great way to reuse locators across tests using `getters` and `methods`.
+- They make it easier to write tests that are more readable and maintainable.
+- Always make sure to use the most restricted scope possible - not the whole page to avoid multiple matches.
+
+If you are writing a new test, make sure to look at the existing page-objects to see if there is a suitable one for your use case.
+
+If not, create the ones you need in the respective context.
```

Comment:
I think not all locators are created equal. Some cases are so specific that will never be reutilized, I'm not sure we should have a "hard" best practice for page object getters creation